srtcore/group.cpp

```diff
@@ -2310,13 +2310,19 @@ int CUDTGroup::recv(char* buf, int len, SRT_MSGCTRL& w_mc)
                 }
             }
         }
+        bool canReadFurther = false;
         for (vector<CUDTSocket*>::const_iterator si = aliveMembers.begin(); si != aliveMembers.end(); ++si)
         {
             CUDTSocket* ps = *si;
             if (!ps->core().isRcvBufferReady())
                 m_Global.m_EPoll.update_events(ps->m_SocketID, ps->core().m_sPollID, SRT_EPOLL_IN, false);
+            else
+                canReadFurther = true;
```

Comment:
Socket being read-ready may be not enough to ensure that the group will be read-ready. For example, if a socket is about to deliver a packet with an old sequence, this isn't a packet that would be next read and delivered from the group reader.

Reply:
In the loop above this one the `rcvDropTooLateUpTo` is called on each member socket to align buffer positions.
I don't see any reason to keep those two loops separated though. Makes sense to combine into a single one.

Reply:
Yeah, that came to my mind, too. There's a scoped lock in the first one, but I don't think it complicates things.

Reply:
OTOH maybe an improvement can be made for the buffer info extraction to provide the information about a certain sequence of a packet that could be available, not just the first ready one. This way you might have the information about potential delivery readiness for every link just at hand.